jenkins/bootstrap.py

```diff
@@ -194,24 +214,43 @@ def upload_artifacts(self, path, artifacts):
 
 def append_result(gsutil, path, build, version, passed):
     """Download a json list and append metadata about this build to it."""
-    # TODO(fejta): ensure the object has not changed since downloading.
     # TODO(fejta): delete the clone of this logic in upload-to-gcs.sh
     #                  (this is update_job_result_cache)
-    cmd = ['gsutil', '-q', 'cat', path]
-    try:
-        cache = json.loads(call(cmd, output=True))
-        if not isinstance(cache, list):
-            raise ValueError(cache)
-    except (subprocess.CalledProcessError, ValueError):
-        cache = []
-    cache.append({
-        'version': version,
-        'buildnumber': build,
-        'passed': bool(passed),
-        'result': 'SUCCESS' if passed else 'FAILURE',
-    })
-    cache = cache[-200:]
-    gsutil.upload_json(path, cache)
+    end = time.time() + 300  # try for up to five minutes
+    errors = 0
+    while time.time() < end:
+        if errors:
+            random_sleep(min(errors, 3))
+        try:
+            out = gsutil.stat(path)
```

Comment:
if `gsutil stat` flakes for some reason (which is not unheard of), won't this cause us to trash the cache?

I guess that's no worse than the current behavior.